test/vec_notation.lean

```diff
@@ -28,3 +30,15 @@ run_cmd Elab.Command.liftTermElabM do
 #guard (toString (repr (![] : _ → ℕ)) = "![]")
 #guard (toString (repr ![1, 2, 3]) = "![1, 2, 3]")
 #guard (toString (repr ![![1, 2], ![3, 4]]) = "![![1, 2], ![3, 4]]")
+
+syntax "#guard_delab" term:51 " :→ " term : command
+
+elab_rules : command
+  | `(command| #guard_delab $t₁ :→ $t₂) =>
+    Command.liftTermElabM <| Term.withDeclName `_guard_delab <| do
+      let e₁ ← Term.elabTerm t₁ none
+      let t₁' ← delab e₁
+      guard <| t₁' == t₂
+
+#guard_delab fun x => (![0, 1] : Fin 2 → ℕ) x :→ fun x => ![0, 1] x
```

Comment:
Let's use `#guard_msgs` instead of using a custom testing command.

I didn't test this (I'm guessing the docstring), and you'll probably have to use the code action for `#guard_msgs` to update it.
```suggestion
/-- info: fun x => ![0, 1] x : Fin 2 → ℕ -/
#guard_msgs in #check fun x => (![0, 1] : Fin 2 → ℕ) x
```

(By the way, there are some bugs in `#guard_delab`. The first line should be something like `let e₁ ← instantiateMVars <| ← withSynthesize <| Term.elabTerm t₁ none`, but it's better that we don't spend time debugging it and instead use the pre-existing tool.)

Reply:
I think you also should set `set_option pp.unicode.fun false` at the top of this file so that the test file runs the same both in the IDE and in CI.